modules/python/src2/cv2.cpp

```diff
@@ -1111,17 +1102,17 @@ template<typename _Tp> struct pyopencvVecConverter
                     return false;
                 data[j] = saturate_cast<_Cp>(v);
             }
-            else if( PyLong_Check(item_ij))
+            else if( PyFloat_Check(item_ij))
             {
-                int v = (int)PyLong_AsLong(item_ij);
-                if( v == -1 && PyErr_Occurred() )
+                double v = PyFloat_AsDouble(item_ij);
+                if( PyErr_Occurred() )
                     return false;
                 data[j] = saturate_cast<_Cp>(v);
             }
-            else if( PyFloat_Check(item_ij))
+            else if( PyLong_Check(item_ij) || PyNumber_Check(item_ij))
             {
-                double v = PyFloat_AsDouble(item_ij);
-                if( PyErr_Occurred() )
+                int v = (int)PyLong_AsLong(item_ij);
+                if( v == -1 && PyErr_Occurred() )
```

Comment:
Can you please check the final version of changes? Is that necessary order of conditions?